message/validation/validation.go

```diff
@@ -96,6 +97,12 @@ func (mv *messageValidator) Validate(_ context.Context, peerID peer.ID, pmsg *pu
 
 	decodedMessage, err := mv.handlePubsubMessage(pmsg, time.Now())
 	if err != nil {
+		if mv.netCfg.Beacon.EstimatedCurrentSlot() == mv.netCfg.Beacon.FirstSlotAtEpoch(mv.netCfg.AlanForkEpoch) {
+			fullErr, ok := err.(Error)
+			if ok && errors.Is(fullErr.innerErr, ErrMalformedPubSubMessage) {
```

Comment:
`innerErr` never contains ErrMalformedPubSubMessage, it has an underlying error. We need to check if `fullErr.text == ErrMalformedPubSubMessage.text`. I wouldn't say I like that we cannot use a single `errors.Is/As` check but this is how errors are designed now. Perhaps, we could consider a better solution in the future.